### PR TITLE
panda proxy features

### PR DESCRIPTION
**Pull request by agbogdan**
pull request: all features concerning panda proxy (get/update event ranges and OS file uploading)

### CODE REVIEW

File: EventRangesPandaProxy.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+#
+
+import json
+import os
+import requests
```
Comment (PalNilsson) [inline EventRangesPandaProxy.py]: requests is not a standard python module. Please put a try statement around it so it doesn't crash the code if the module gets imported on a site where this module is not available.

File: PILOTVERSION [hunk 2]
```diff
@@ -1,2 +1,2 @@
-PICARD 69.0 RC7 BUILD3
```
Comment (PalNilsson) [inline PILOTVERSION]: You don't need to update this file (it leads to conflicts when I make a separate commit)

File: S3ObjectstorePresignedURLSiteMover.py [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+        self.length = len(iterable)
+
+    def read(self, size=-1):   # TBD: add buffer for `len(data) > size` case
+        return next(self.iterator, b'')
+
+    def __len__(self):
+        return self.length
+
+
+class S3ObjectstorePresignedURLSiteMover(SiteMover.SiteMover):
```
Comment (PalNilsson) [inline S3ObjectstorePresignedURLSiteMover.py]: This class inherits from the old site movers which are quickly becoming deprecated now. How will this mover be used, only on BOINC? Nevertheless it will depend on deprecated code. We should think about how/if this can be solved with the new movers. Can you please discuss this with Wen Guan and Alexey Anisenkov? It could be that we only need to worry about new movers once we can implement all of this new functionality in Pilot 2, but please discuss it with Wen and Alexey.

Comment (PalNilsson) [inline S3ObjectstorePresignedURLSiteMover.py]: Ah I see your other pandaproxy_sitemover code now, so you support both old and new movers?

File: movers/pandaproxy_sitemover.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+
+#from pUtil import getSiteInformation
+
+from PilotErrors import PilotException
+#from PilotErrors import PilotErrors
+
+import os
+#import traceback
+import requests
```
Comment (PalNilsson) [inline movers/pandaproxy_sitemover.py]: Same comment as above. Be careful with how this module is imported and used.

**Comment by PalNilsson** [comment]
Thanks for the pull request. Please see the inline comments.

File: EventRangesPandaProxy.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    tolog("Updating event ranges..")
+
+    for range in event_ranges:
+        eventRangeId = range['eventRangeID']
+        eventRangeStatus = range['eventStatus']
+        retStatus, message =  updateEventRangePandaProxy_normalized(eventRangeId, jobId, pandaProxySecretKey, eventRangeStatus)
+        if message != "":
+                return retStatus, mesage
+    return retStatus, message
+
```
Comment (PalNilsson) [inline EventRangesPandaProxy.py]: Make sure there's no bad character inserted in this line

File: S3ObjectstorePresignedURLSiteMover.py (hunk 3 above)
Comment (agbogdan) [inline S3ObjectstorePresignedURLSiteMover.py]: The S3ObjectstorePresignedURLSiteMover class contains the latest code of the pandapoxy old sitemover (started by Wen as S3ObjectstoreHttpSiteMover).
It provides "stage-in from objectstore with panda proxy" functionality, which was not demanded by BOINC and was not implemented in the new sitemover. May be it will be useful as an example.

**Comment by PalNilsson** [comment]
Ok thanks.
